src/Core/Managed/Shared/Extensibility/MetricManager.cs

```diff
@@ -119,12 +119,6 @@ public void Flush()
         /// </summary>
         public void Dispose()
         {
-            if (this.snapshotTimer != null)
-            {
-                this.snapshotTimer.Dispose();
-                this.snapshotTimer = null;
-            }
-
             this.Flush();
```

Comment:
I tend to agree, but it is still a separate issue. Once this is approved, I will complete the PR and then you have make that change as a part of the general MetricsManager dev work. Would that be OK?